byte_zero.c

```diff
@@ -1,13 +1,8 @@
 #include "byte.h"
 
-void byte_zero(s,n)
-char *s;
-register unsigned int n;
+void byte_zero(char *s, unsigned int n)
```

Comment:
Is this not to be replaced by memset so we can remove this file entirely?

```
#define byte_zero(s, n) memset(s, 0, n)
```

Reply:
I intentionally kept this so the compiler will not optimize the zeroing aways. I need a lot of logic to get this portable in Qsmtp, and there is still the possibility that it will fallback to not finding one: https://github.com/DerDakon/Qsmtp/blob/master/qsmtpd/auth.c#L29

Reply:
Would a comment be welcome on the source so that we state out where we want *not* to edit it?  Although maybe `byte_zero` is a famous enough topic.

https://github.com/libressl-portable/openbsd/issues/5

It looks like there is a hard time getting this not optimized away.

Reply:
Our luck is that we don't use -flto at the moment. And the other thing is that it is used in that "dangerous" way only in qmail-popup, which will hopefully just go away together with qmail-pop3d.

Reply:
Cross-referencing #65 which at one point was suggesting

```c
#define byte_zero(s, n) memset((s), 0, (n))
```

now that I see this conversation here :-)

(Hopefully not derailing this PR: I know in our last release we expressed intent to delete `qmail-popup` and `qmail-pop3d`, but I'm not ready to do that yet, even though I don't run them myself. They're a worked example of a particularly qmail-ish design we might want to follow for AUTH and extend to support TLS, so I'd like us to hang onto them at least until we've added those features -- assuming they're relatively cheap to carry with us a bit longer, which I think is true.)